app/screens/crisp-support-screen/crisp-support-screen.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react"
+import { ViewStyle } from "react-native"
+import { NavigationScreenProps } from "react-navigation"
+import { WebView } from "react-native-webview"
+import { inject, observer } from "mobx-react"
+
+import { RootStore } from "../../models/root-store"
+import { UserStore } from "../../models/user-store"
+
+import { COMMON_API_CONFIG } from "../../services/api/api-config"
+
+import { Header } from "../../components/header"
+import { Screen } from "../../components/screen"
+
+import { color } from "../../theme"
+
+const FULL: ViewStyle = { flex: 1 }
+
+export interface CrispSupportScreenProps extends NavigationScreenProps<{}> {
+  userStore: UserStore
+  CRISP_WEBSITE_ID?: string
+}
+
+@inject((allStores: any) => ({
+  userStore: allStores.userStore as UserStore,
+  CRISP_WEBSITE_ID: (allStores.rootStore as RootStore).env.appConfig.getValue("CRISP_WEBSITE_ID"),
+}))
+@observer
+export class CrispSupportScreen extends React.Component<CrispSupportScreenProps, {}> {
+  private goBack = () => {
+    this.props.navigation.goBack()
+  }
+
+  render() {
+    const { CRISP_WEBSITE_ID } = this.props
+    const { currentUser } = this.props.userStore
+    const { profile } = this.props.userStore.authCore
+    let uri = CRISP_WEBSITE_ID ? `https://go.crisp.chat/chat/embed/?website_id=${CRISP_WEBSITE_ID}` : 'https://help.like.co'
+    if (CRISP_WEBSITE_ID) {
+      if (currentUser && currentUser.email) uri += `&email=${encodeURIComponent(currentUser.email)}`
+      if (profile && profile.primaryPhone) uri += `&phone=${encodeURIComponent(profile.primaryPhone)}`
+    }
```

Comment:
I think u can add a view property in the `userStore` to return this URL, and remove the `CRISP_WEBSITE_ID` props as well

```javascript
...
get crispChatEmbedURL() {
   ...
   return url
}
...
```

Reply:
but crisp website id seems to have nth to do with user, do any other useful store exists?

Reply:
As I saw the URL is depending on the user's email & the AuthCore's profile, so I suggest using computed property in MST to compose the URL, also make the `render()` looks cleaner

Reply:
Get this straight
```tsx
// user-store.ts
...
get crsipChatEmbedURL() {
  const crispID = self.env.appConfig.getValue("CRISP_WEBSITE_ID")
  if (crispID) {
    const { profile } = self.authCore
    let uri = `https://go.crisp.chat/chat/embed/?website_id=${crispID}`
    if (self.currentUser?.email) {
      uri += `&email=${encodeURIComponent(self.currentUser.email)}`
    }
    if (profile?.primaryPhone) {
      uri += `&phone=${encodeURIComponent(profile.primaryPhone)}`
    }
  }
  return "https://help.like.co"
}
...

// crisp-support-screen.tsx
<WebView source={{ uri: this.props.userStore.crsipChatEmbedURL }} {...} />
```

Reply:
Get this straight 
```tsx
// user-store.ts
...
get crispChatEmbeddedURL() {
  const crispID = self.env.appConfig.getValue("CRISP_WEBSITE_ID")
  let uri: string
  if (crispID) {
    const { profile } = self.authCore
    uri = `https://go.crisp.chat/chat/embed/?website_id=${crispID}`
    if (self.currentUser?.email) {
      uri += `&email=${encodeURIComponent(self.currentUser.email)}`
    }
    if (profile?.primaryPhone) {
      uri += `&phone=${encodeURIComponent(profile.primaryPhone)}`
    }
  } else {
    uri = "https://help.like.co"
  } 
  return uri
}
...

// crisp-support-screen.tsx
<WebView source={{ uri: this.props.userStore.crispChatEmbeddedURL }} {...} />
```